src/marshmallow_union/__init__.py

```diff
@@ -59,9 +59,16 @@ def _serialize(self, value: t.Any, attr: str, obj: str, **kwargs):
         for candidate_field in fields:
 
             try:
-                return candidate_field.serialize(
-                    attr, obj, error_store=error_store, **kwargs
-                )
+                try:
+                    return candidate_field.serialize(
+                        attr, obj, error_store=error_store, **kwargs
+                    )
+                except TypeError:
+                    if attr is obj is None:  # is a mapping value
```

Comment:
A longer comment explaining what's going on here would be helpful for later when I forget about this :-).